File: app/src/main/java/cn/yue/base/kotlin/test/AppApplication.java
package cn.yue.base.kotlin.test;

import android.app.Application;
import android.content.Context;

import com.alibaba.android.arouter.launcher.ARouter;

import cn.yue.base.common.utils.NetworkUtils;

/**
 * Description :
 * Created by yue on 2018/11/14
 */
public class AppApplication extends Application{

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
//        MultiDex.install(this);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        ARouter.init(this);
        ARouter.openDebug();
        NetworkUtils.setContext(this);
    }
}

File: app/src/main/kotlin/cn/yue/base/kotlin/test/AppApplication.kt
package cn.yue.base.kotlin.test

import cn.yue.base.kotlin.test.utils.LocalStorage
import cn.yue.base.middle.init.InitConstant
import cn.yue.base.middle.init.MiddleApplication
import cn.yue.base.middle.module.IAppModule
import cn.yue.base.middle.module.ModuleType
import cn.yue.base.middle.module.manager.ModuleManager

/**
 * Description :
 * Created by yue on 2018/11/14
 */
class AppApplication : MiddleApplication() {

    override fun init() {
        InitConstant.setDebug(BuildConfig.DEBUG_MODE)
        InitConstant.setServiceEnvironment(LocalStorage.getServiceEnvironment())
        InitConstant.setVersionName(BuildConfig.VERSION_NAME)
        super.init()
    }

    override fun registerModule() {
        ModuleManager.register(ModuleType.MODULE_APP, IAppModule::class, AppModuleService())
    }
}